plasTeX/ConfigManager.py

```diff
@@ -0,0 +1,181 @@
+from argparse import _ArgumentGroup as ArgumentGroup, ArgumentParser
+from abc import ABC, abstractmethod
+from typing import Dict, List, Generic, TypeVar, Any, Optional, Type, KeysView, Union
+from configparser import ConfigParser
+import shlex
+
+T = TypeVar('T')
+
+def formatDefault(s: Any):
+    return " [" + str(s).replace("%", "%%") + "]"
+
+class Option(Generic[T]):
+    def __init__(self, description: str, options: str, default: T):
+        self.description = description
+        self.options = options.split(" ")
+        self.name = self.options[0].lstrip("-")
+        self._value = default
+
+    def registerArgparse(self, group: ArgumentGroup):
+        group.add_argument(*self.options, dest=self.name, type=self.valueType(), help=self.description + formatDefault(self.value))
+
+    def valueType(self) -> Type:
+        return type(self.value)
+
+    def updateFromDict(self, data: Dict["str", Any]):
+        value = data.get(self.name)
+        if value is not None:
+            self.value = value
+
+    def setFromString(self, string: str):
+        self.value = self.valueType()(string)
+
+    @property
+    def value(self) -> T:
+        return self._value
+
+    @value.setter
+    def value(self, value: T):
+        if type(value) != self.valueType():
+            raise TypeError("Encountered {} when expecting {}".format(type(value).__name__, self.valueType().__name__))
+
+        self._value = value
+
+class BooleanOption(Option[bool]):
```

Comment:
True/False works, but yes/no does not. A string is converted to a boolean via bool(). I can add them back if you wish.

Reply:
I really don't know. *I* certainly don't need it. And it clearly goes back very far in time, at a time where `bool` was not properly handled in python. Also it's really annoying that we don't have any information about how plasTeX is used in the wild.

I would say we don't need to keep backward compatibility, but it would be nice to have a changelog listing those user-facing changes.